tests/test_process_elife_zip_dag.py

```diff
@@ -45,60 +48,153 @@ def test_get_expected_elife_article_name_raises_exception_if_zip_name_is_malform
         assert str(error.value) == msg
 
 
+def test_get_article_from_previous_task(context):
+    # setup
+    test_asset_path = str(get_asset('elife-00666.xml').absolute())
+    article_xml = etree.tostring(etree.parse(test_asset_path))
+    add_return_value_from_previous_task(return_value=article_xml, context=context)
+    # test
+    returned_value = get_article_from_previous_task(context)
+    assert etree.tostring(returned_value) == article_xml
+
+
+@pytest.mark.parametrize('return_value', [b'', '', None])
+def test_get_article_from_previous_task_raises_exception(return_value, context):
+    # setup
+    add_return_value_from_previous_task(return_value, context)
+    message = 'Article bytes were not passed from task previous_task'
+    # setup
+    with pytest.raises(AssertionError) as error:
+        get_article_from_previous_task(context)
+        assert str(error.value) == message
+
+
 def test_extract_archived_files_to_bucket(context, s3_client):
+    # setup
     file_name = 'elife-00666-vor-r1.zip'
     context['dag_run'].conf = {'file': file_name}
+    # test
     extract_archived_files_to_bucket(**context)
     for zipped_file in ZipFile(get_asset(file_name)).namelist():
         expected_file = '%s/%s' % (file_name.replace('.zip', ''), zipped_file)
         assert expected_file in s3_client.uploaded_files
 
 
 def test_extract_archived_files_to_bucket_raises_exception_when_article_not_in_zip(context, mocker, s3_client):
+    # setup
     context['dag_run'].conf = {'file': 'elife-00666-vor-r1.zip'}
     mocker.patch('zipfile.ZipFile.namelist', return_value=[])
+    # test
     with pytest.raises(FileNotFoundError) as error:
         extract_archived_files_to_bucket(**context)
         assert str(error.value) == 'elife-00666.xml not in elife-00666-vor-r1.zip: []'
 
 
-def test_convert_tiff_images_in_expanded_bucket_to_jpeg_images(context, s3_client, mocker):
+def test_convert_tiff_images_in_expanded_bucket_to_jpeg_images_using_article_with_tiff_images(context, s3_client, mocker):
+    # setup
     file_name = 'elife-36842-vor-r3.zip'
-    folder_name = file_name.replace('.zip', '/')
     context['dag_run'].conf = {'file': file_name}
+
+    folder_name = file_name.replace('.zip', '/')
     keys = [folder_name + fn for fn in ZipFile(get_asset(file_name)).namelist()]
-    keys.append(folder_name)
+    keys = itertools.chain(keys, [folder_name])
     mocker.patch('dags.process_elife_zip_dag.list_bucket_keys_iter', return_value=keys)
 
+    # test
     convert_tiff_images_in_expanded_bucket_to_jpeg_images(**context)
+
+    # recreate expected uploaded file names
     zipped_files = [fn.replace('.tif', '.jpg')
```

Comment:
a bit involved to have to calculate a complex expected value, what about relaxing these assertions and check just the number of files, plus a couple of examples from `s3_client.uploaded_files`?

Reply:
yeah sure. I'll update.